primerj/src/main/java/org/primer/primerj/core/TxBuilder.java

```diff
@@ -821,9 +826,8 @@ public Tx buildTx(String changeAddress, List<Out> unspendOuts, Tx tx, Coin... co
         while (true) {
             long fees = 0;
 
-            if (lastCalculatedSize >= 1000) {
-                // If the size is exactly 1000 bytes then we'll over-pay, but this should be rare.
-                fees += (lastCalculatedSize / 1000 + 1) * Utils.getFeeBase();
+            if (lastCalculatedSize >= 0) {
+                fees += lastCalculatedSize * (Utils.getFeeBase() / 1000);
             }
             if (needAtLeastReferenceFee && fees < Utils.getFeeBase())
                 fees = Utils.getFeeBase();
```

Comment:
similarly, consider removal of these two lines block if (needAtLeastReferenceFee && ...) {}

Reply:
similarly, consider removal of line 847-861

Reply:
line 904: if (size / 1000 > lastCalculatedSize / 1000 && Utils.getFeeBase() > 0)
similarly, this condition may need to be updated